src/util/time.h

```diff
@@ -22,6 +22,15 @@ struct NodeClock : public std::chrono::system_clock {
 };
 using NodeSeconds = std::chrono::time_point<NodeClock, std::chrono::seconds>;
 
+/** Mockable steady clock in the context of tests, otherwise the system steady clock */
+struct NodeSteadyClock : public std::chrono::steady_clock {
```

Comment:
Could we turn `SteadyClock` into this struct instead? Having both types could be confusing.

Reply:
sure, fine with me, agree it's less confusing though i wasn't sure which of the current usages should be mockable and didn't want to decide that here

Reply:
>  i wasn't sure which of the current usages should be mockable and didn't want to decide that here

fair enough!

`MockableSteadyClock` sgtm!

Reply:
Should be unambigious now, i've renamed it and the methods to set/clear the mock time are included on the clock itself.
```c++
/**
 * Version of SteadyClock that is mockable in the context of tests (set the
 * current value with SetMockSteadyTime), otherwise the system steady clock.
 */
struct MockableSteadyClock : public std::chrono::steady_clock {
    using time_point = std::chrono::time_point<MockableSteadyClock>;
    /** Return current system time or mocked time, if set */
    static time_point now() noexcept;
    static std::time_t to_time_t(const time_point&) = delete; // unused
    static time_point from_time_t(std::time_t) = delete;      // unused

    /** Set mock time for testing. */
    static void SetMockTime(std::chrono::milliseconds mock_time_in);

    /** Clear mock time, go back to system steady clock. */
    static void ClearMockTime();
};
```